index.d.ts

```diff
@@ -0,0 +1,9 @@
+// TypeScript Version: 3.5.2
+
+interface Options {
+  limit: number;
+}
+
+declare function safeRegex<T>(regex: RegExp, opts?: Options | T): void;
```

Comment:
- currently the safeRegex function takes `RegExp`'s ,`strings` and anything with a toString method in its re parameter (so technically `any`, I guess). Which'd make this declaration a bit too restrictive.
- this definition seems to have `void` as a return type - but the function currently returns a boolean (A void return type would've been correct if safeRegex would've been implemented as a function that threw an exception in case of problems and silently exited in all other cases).
- I see you're using a [_generic_](https://www.typescriptlang.org/docs/handbook/generics.html) (the `<T>` & `T` in the declaration - had to look that one up) why is that?